server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -40,14 +40,78 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
 
         var httpClient = HttpClient.newHttpClient();
         var youtubeReq = HttpRequest.newBuilder(youtubeUri)
-                .header("Authorization", "Bearer " + accessToken.toString())
+                .header("Authorization", "Bearer " + accessToken)
                 .header("Accept", "application/json")
                 .GET()
                 .build();
 
         // get response to Youtube API 
         var youtubeRes = httpClient.sendAsync(youtubeReq, BodyHandlers.ofString()).join();
-        var youtubeResBody = youtubeRes.body();
-        res.getWriter().write(youtubeResBody);
+        return youtubeRes.body();
+    }
+
+    /**
+     * checks whether topic is categorized as music
+     * by checking if the last word is "music" or "Music"
+     * @param topic 
+     * @return Boolean
+     */
+    protected Boolean isMusic(String topic) {
+        String lastWord = topic.substring(topic.lastIndexOf(" ") + 1);
+        return lastWord.equals("Music") || lastWord.equals("music");
+    }
+
+    /**
+     * parses through youtube liked videos json string,
+     * updates hash map to contain frequency count of each music genre
+     * @param youtubeResBody json response of youtube liked videos
+     * @param genreCount hash map of frequency count of each music genre
+     */
+    protected void updateMusicCount(String youtubeResBody, HashMap<String, Integer> genreCount) {
+        JsonObject jObject = JsonParser.parseString(youtubeResBody).getAsJsonObject();
+        JsonArray videos = jObject.getAsJsonArray("items");
+
+        // iterates through each liked video
+        for (int i = 0; i < videos.size(); i++) {
+            // extracts array of topicCategories in video
+            JsonObject video = videos.get(i).getAsJsonObject();
+            JsonObject topicDetails = video.getAsJsonObject("topicDetails");
+            JsonArray topicCategories = topicDetails.getAsJsonArray("topicCategories");
```

Comment:
Advice? I'm going from items > topicDetails > topicCategories in JSON object. This works if json response works as it should, but probably need to error check each step of the way.

Reply:
well, the YouTube API is not likely to just change suddenly. since this is not user-generated data, you can probably get away with not error-checking it; even if an exception is thrown it just turns the request into a 500 error. the most Correct™️ thing to do though is to check each of these JSON objects before using them and return a 400 error if any of them are `null`.

Reply:
@ everyone else I'm going to deal with case where video does not have topicDetails json in an upcoming branch `youtube-merge-error`